modules/imgcodecs/src/ios_conversions.mm

```diff
@@ -116,6 +118,8 @@ void UIImageToMat(const UIImage* image,
         if (!alphaExist)
             bitmapInfo = kCGImageAlphaNoneSkipLast |
                                 kCGBitmapByteOrderDefault;
+        else
+            m = cv::Scalar(0);
```

Comment:
Oh I see now.
May be it make sense to initialize the Mat for other branch as well to ensure the alpha channel doesn't contain garbage. As far as I understand it wouldn't be changed during CGContextDrawImage execution.